pkg/types/baremetal/validation/platform.go

```diff
@@ -35,14 +39,63 @@ func ValidatePlatform(p *baremetal.Platform, fldPath *field.Path) field.ErrorLis
 
 	if err := validate.IP(p.APIVIP); err != nil {
 		allErrs = append(allErrs, field.Invalid(fldPath.Child("apiVIP"), p.APIVIP, err.Error()))
+	} else if n != nil {
```

Comment:
nit: can we drop the `else if` and move it to just `if`.

Reply:
Sure.